scripts/prepare_int_data.py

```diff
@@ -0,0 +1,185 @@
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + "/../src");
+
+# do some checking
+if os.system("validate_text_dir.py " + args.text_dir) != 0:
+  sys.exit("prepare_int_data.py: failed to validate text directory")
+
+if os.system("validate_vocab.py " + args.vocab) != 0:
+  sys.exit("prepare_int_data.py: failed to validate vocab")
+
+if not os.path.exists(os.path.abspath(os.path.dirname(sys.argv[0])) + "/text_to_int.py"):
+  sys.exit("prepare_int_data.py: expected text_to_int.py to be on the path")
+
+# create the output data directory
+if not os.path.exists(args.int_dir):
+  os.makedirs(args.int_dir)
+
+if not os.path.exists(args.int_dir + "/log"):
+  os.makedirs(args.int_dir + "/log")
+
+# remove any old *.int.gz files in the output data directory
+filelist = [f for f in os.listdir(args.int_dir) if f.endswith(".int.gz")]
+for f in filelist:
+  try:
+    os.remove(f)
+  except OSError:
+    pass
+
+def ExitProgram(message):
+  print(message, file=sys.stderr)
+  os._exit(1)
+
+def RunCommand(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  if os.system(command) != 0:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+
+def GetCommandStdout(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  try:
+    output = subprocess.check_output(command, shell = True)
+  except:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+  return output
+
+def GetNames(script_dir, text_dir, int_dir):
+  command = "{0}/internal/get_names.py {1} > {2}/names".format(
+              script_dir, text_dir, int_dir) 
+  RunCommand(command)
+
+def GetNumTrainSets(int_dir):
+  command = "cat {0}/names | wc -l".format(int_dir)
+  line = subprocess.check_output(command, shell = True) 
+  try:
+    a = line.split()
+    assert len(a) == 1
+    ans = int(a[0])
+  except:
+    sys.exit("prepare_int_data.py: error: unexpected output '{0}' from command {1}".format(
+                line, command))
+  return ans
+
+def CopyFile(src, dest):
+  try:
+    shutil.copyfile(src, dest)
+  except: 
+    sys.exit("prepare_int_data.py: error copying {0} to {1}".format(src, dest))
+
+# returns num-words in vocab.
+def GetNumWords(vocab):
+  command = "tail -n 1 {0}".format(vocab)
+  line = subprocess.check_output(command, shell = True)
+  try:
+    a = line.split()
+    assert len(a) == 2
+    ans = int(a[1])
+  except:
+    sys.exit("prepare_int_data.py: error: unexpected output '{0}' from command {1}".format(
+                line, command))
+  return ans
+
+# we can include the preparation of the dev data in the following
+# by adding "dev dev" to the contents of int_dir/names
+def PrepareData(line):
+  [int, name] = line.split()
+  if os.path.exists(args.text_dir + "/" + name + ".txt.gz"):
+    command = "set -o pipefail; gunzip -c {text_dir}/{name}.txt.gz | text_to_int.py {vocab} | gzip -c > {int_dir}/{int}.txt.gz 2>{int_dir}/log/{int}.log".format(
```

Comment:
Just modified the code based on your comments.
Ke

On Sat, Jun 25, 2016 at 6:52 PM, Daniel Povey <notifications@github.com>
wrote:

> In scripts/prepare_int_data.py
> <https://github.com/danpovey/pocolm/pull/18#discussion_r68495029>:
>
> > +  line = subprocess.check_output(command, shell = True)
> > +  try:
> > +    a = line.split()
> > +    assert len(a) == 2
> > +    ans = int(a[1])
> > +  except:
> > +    sys.exit("prepare_int_data.py: error: unexpected output '{0}' from command {1}".format(
> > +                line, command))
> > +  return ans
> > +
> > +# we can include the preparation of the dev data in the following
> > +# by adding "dev dev" to the contents of int_dir/names
> > +def PrepareData(line):
> > +  [int, name] = line.split()
> > +  if os.path.exists(args.text_dir + "/" + name + ".txt.gz"):
> > +    command = "set -o pipefail; gunzip -c {text_dir}/{name}.txt.gz | text_to_int.py {vocab} | gzip -c > {int_dir}/{int}.txt.gz 2>{int_dir}/log/{int}.log".format(
>
> please break this long line, e.g. by using backslash \ or parentheses.
> Likewise else there. Don't go too far past 80 characters.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/18/files/b4e574c7a96866d2347764cc620ecaaddd049c5c#r68495029>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANVxSkHbGPyNTr9qN1aw3T3OxM-yE-ofks5qPbEYgaJpZM4I-PTe>
> .
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu